src/rules/no-extraneous-dependencies.js

```diff
@@ -196,7 +219,7 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
 
     if (
       declarationStatus.isInDeps ||
-      (depsOptions.allowDevDeps && declarationStatus.isInDevDeps) ||
+      ((depsOptions.allowDevDeps || isTypeOnly) && declarationStatus.isInDevDeps) ||
```

Comment:
It's common practice to declare type-only dependencies in devDependencies -- that was among the reasons for originally ignoring type-only imports: #1618